docs/apm/traces/get-started-transaction-tracing/opentelemetry-instrumentation/net/traceid-spanid-injection-into-logs.md

```diff
@@ -36,10 +38,24 @@ private static IHostBuilder CreateHostBuilder(string[] args) =>
             });
 ```
 
-:::note
-The above solution has not yet been tested internally.
-:::
+# ASP.NET Logs4net
 
+1. Get the current traceId and spanId, which can be obtained by:
+```cs
+log4net.ThreadContext.Properties["trace_id"] = Tracer.CurrentSpan.Context.TraceId;
+log4net.ThreadContext.Properties["span_id"] = Tracer.CurrentSpan.Context.SpanId;
+```
+
+2. Extend log pattern layout in log4net.config file, like in attached configuration with
+two additional properties `%property{trace_id}` and `%property{span_id}`
+```cs
+<appender name="textFileAppender" type="log4net.Appender.FileAppender">
+            <file value="log-log4net-textFile.log" />
+            <layout type="log4net.Layout.PatternLayout">
+                <conversionPattern value="%date [%thread] %level %logger  %property{trace_id} %property{span_id} - %message%newline" />
```

Comment:
According to our formatting rules, this line should be `<conversionPattern value="%date [%thread] %level %logger  trace_id=%property{trace_id} span_id=%property{span_id} - %message%newline" />`

Reply:
Corrected